kanren/cons.py

```diff
@@ -0,0 +1,243 @@
+from functools import reduce
+from itertools import chain
+from operator import length_hint
+from collections import OrderedDict
+from collections.abc import Iterator
+
+from multipledispatch import dispatch
+
+from toolz import drop, last
+
+from unification.core import unify, _unify, reify, _reify
+
+
+def first(x):
+    return next(iter(x), None)
+
+
+def rest(x):
+    return drop(1, x)
+
+
+def none_to_list(x):
+    return [] if x is None else x
+
+
+@dispatch(object, object)
+def cons_merge(car_part, cdr_part):
+    """Merge a generic car and cdr."""
+    return None
```

Comment:
Is this a TODO? It's unclear to me how this method body does what the function name or docstring say they're doing.

Reply:
No, not a TODO; that's supposed to be the fall-back/base case.

Reply:
I guess that having the generic function's docstring there is a bit misleading; will remove it.

Reply:
Maybe I don't fully understand what this is used for. It seems like either you're saying:
 1. Returning `None` is _correct_ when arbitrary types are passed (not explicitly handled below) as a base case to the recursion.
 2. This is required by the dispatch library but won't actually be called in practice.

If (1), it's at least somewhat nonobvious at first glance why that's the case and could use some comments explaining it. If it's (2), then it seems like raising a `TypeError` would be preferable.

Reply:
It's (2); I'm looking into removing it and `none_to_list` entirely.

Reply:
Done.  Also, `None` is now allowed as a `car` without being turned into a list (not sure why I was doing that).